lib/email/receiver.rb

```diff
@@ -765,23 +765,44 @@ def create_group_post(group, user, body, elided)
           .order(created_at: :desc)
           .limit(1)
           .pluck(:post_id).last
-        post_ids << post_id_from_email_log
+        post_ids << post_id_from_email_log if post_id_from_email_log
       end
 
       if post_ids.any? && post = Post.where(id: post_ids).order(:created_at).last
 
-        # this must be done for the unknown user (who is staged) to
-        # be allowed to post a reply in the topic
-        if group.allow_unknown_sender_topic_replies
-          post.topic.topic_allowed_users.find_or_create_by!(user_id: user.id)
-        end
+        # We don't want users reopening months-old PM discussions in
+        # group SMTP inbox environments. Instead we just create a new
+        # topic with a link back to the old one.
+        if destination_too_old?(post) && group.smtp_enabled
```

Comment:
Logic to create new topic in this branch `if destination_too_old?(post) && group.smtp_enabled` and `else` is very similar (there is a difference in raw content).

What do you think if we merge them? Something like:
```ruby 
last_post = post_ids.any? && post = Post.where(id: post_ids).order(:created_at).last
if !post || (destination_too_old?(post) && group.smtp_enabled)
  enable_email_pm_setting(user)
  create_topic # with different raw if necessary, maybe define method to calculate raw
else 
  create_reply
end
```

Reply:
Yeah good point about the similarity...will clean it up.